tide_site_restriction.module

```diff
@@ -41,6 +41,10 @@ function tide_site_restriction_entity_access(EntityInterface $entity, $operation
     return AccessResult::neutral()->addCacheableDependency($entity);
   }
   if ($account->isAuthenticated() && in_array($operation, ['update', 'delete'])) {
+    // Restricting editors from deleting contents SDPA-3781.
+    if (in_array('editor', $account->getRoles()) && $operation == 'delete') {
```

Comment:
@vincent-gao Because of this condition` if ($account->isAuthenticated() && in_array($operation, ['update', 'delete']))` in `tide_site_restriction.module` when the site gets assigned the user gets the permission to update and delete as well. This is the reason I am adding this condition here instead of doing it from the configuaration.

Reply:
@MdNadimHossain Is there a way to avoid hard coding the role `editor`? Something similar to what's validating the user form https://github.com/dpc-sdp/tide_site_restriction/blob/5e639608c448975f9a89873dfb21fa66a5fe98d7/tide_site_restriction.module#L261
```suggestion
  $roles = Role::loadMultiple();
    foreach ($roles as $role_id => $role) {
      if ((!$role->hasPermission('bypass node access') || !$role->hasPermission('administer nodes') || !$role->hasPermission('bypass site restriction')) && $operation == 'delete') {
```
That role-> permission check could definitely be moved out into a helper function.